src/main/kotlin/org/rust/stdext/Collections.kt

```diff
@@ -112,6 +112,11 @@ inline fun <T> List<T>.notEmptyOrLet(function: (List<T>) -> List<T>): List<T> =
     else -> function(this)
 }
 
+inline fun <T> List<T>.notEmptyElseNull(): List<T>? = when {
+    isNotEmpty() -> this
+    else -> null
+}
```

Comment:
You can use the `takeIf` method:

![image](https://user-images.githubusercontent.com/6079006/85050670-19dc3600-b19f-11ea-8865-3760afc42ae7.png)